src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireTokenHandlerBase.cs

```diff
@@ -135,7 +135,19 @@ protected AcquireTokenHandlerBase(RequestData requestData)
 
                     this.NotifyBeforeAccessCache();
                     notifiedBeforeAccessCache = true;
-                    ResultEx = this.tokenCache.LoadFromCache(CacheQueryData, this.CallState);
+
+                    var aliasedAuthorities = await GetOrderedAliases(
+                        GetHost(this.Authenticator.Authority), this.Authenticator.ValidateAuthority, this.CallState).ConfigureAwait(false);
+                    foreach (var aliasedAuthority in aliasedAuthorities)
```

Comment:
this could be moved into TokenCache class.

Reply:
It could. But then it would require the TokenCache to be aware of not just an authority string, but also the entire InstanceCache thing in order to swap different authoiry aliases. While such conceptual difference is not necessarily bad, the substantial change is that `LoadFromCache(...)` would then have to become an async method, and potentially wait arbitrarily long for slow network I/O. It does not sound right for a cache class which is supposed to be fast, non-block, and decisive.

Reply:
As an alternative, in the latest commit I try a different approach. I group this inline loop into a standalone helper `LoadResultExFromCache()` which still sits inside this `AcquireTokenHandlerBase` class, to team up with its pre-existing cousin `StoreResultExToCache()`.

Reply:
this is not an alternative. We clearly discussed the separation of concerns and various aspects. I though we agreed to move this in token cache. Is there a disconnect?